File: codengine-cli/test/fixtures/catalog.ts
// TS task functions for the CLI end-to-end tests.
import type { TaskData } from "codengine-runner-ts";

export const echo = (data: TaskData): TaskData => data;
export const output = (data: TaskData): TaskData => data;
export const pick = (data: TaskData): unknown => data.i;
export const route = (data: TaskData): unknown => data.route;
export const nil = (): null => null;
export const emit = (data: TaskData): TaskData[] =>
  Array.from({ length: Number(data.n) }, (_, i) => ({ i }));

File: codengine-cli/test/fixtures/catalog.py
# Python task functions for the CLI end-to-end tests.
# Natural signatures per the invocation contract (named binding).


def echo(**data):
    return data


def output(**data):
    return data


def pick(i):
    return i


def route(route):
    return route


def nil(**data):
    return None


def emit(n):
    return [{"i": i} for i in range(n)]
